File: producer/go/book.go
package producer

type Book struct {
	Id    string
	Title string
	Name  string
}

func NewBook() *Book {
	return &Book{}
}

func (b *Book) Source(id string) *Book {
	b.Id = id
	b.Title = "title"
	b.Name = "hello"

	return b
}

func (b *Book) Target() string {
	return b.Title
}

File: producer/js/src/book.ts
export interface IBook {
    id: string,
    title: string,
    name: string
}

export class Book implements IBook {
    id: string;
    name: string;
    title: string;

    source(id: string) {
        this.id = id;
        this.title = "title";
        this.name = "name";
        return this
    }

    target() {
        return this.title
    }
}
